test/test_torch.py

```diff
@@ -1242,6 +1242,45 @@ def backward_func(slf, device):
 
         backward_func(self, device)
 
+    @dtypes(*floating_types())
+    @onlyNativeDeviceTypes
```

Comment:
Does `maxunpool` work for dtypes other than floats? Maybe we should test them as well...

Reply:
I tried complex, half, and ints, and they didn't seem to be supported

Reply:
Nope, looks like float and double are the only supported dtypes for both devices. Half is supported by one of the devices, but not the other, so I could add that in if we really want

Reply:
Let's add it just in case as well? Half is used in the context of quantized nns.